core/src/main/java/org/apache/druid/math/expr/vector/VectorProcessors.java

```diff
@@ -851,17 +861,19 @@ public void processIndex(
                 outputNulls[i] = true;
                 return;
               }
-              final boolean bool = Evals.asBoolean(rightInput[i]);
+              final boolean bool = Evals.asBoolean((String) rightInput[i]);
               output[i] = Evals.asLong(bool);
               outputNulls[i] = bool;
               return;
             } else if (rightNull) {
-              final boolean bool = Evals.asBoolean(leftInput[i]);
+              final boolean bool = Evals.asBoolean((String) leftInput[i]);
```

Comment:
Here too, why is it safe to cast?  do we need to check the type first?  Or maybe check the type of the array that we are gotten, so we can amortize the cost of the check?

Reply:
this safe because (for better or for worse) the base classes explicitly wrap their input processors with `CastToTypeVectorProcessor.cast` (which is a no-op if the processor is already the correct type) to ensure that it is being used with the correct type. I don't remember exactly why I did this, presumably because its important for vector processing that everything is the right type so we don't get ugly class cast exceptions, but I can look into if this is actually necessary or not.